From mithlesh@linsyssoft.com  Fri Mar 20 19:56:43 2009
From: Mithlesh Thukral <mithlesh@linsyssoft.com>
Date: Fri, 20 Mar 2009 17:36:44 +0530 (IST)
Subject: Staging: sxg: Fix a warning dump emitted by rtnl_watchdog during LTP tests
To: Greg Kroah-Hartman <gregkh@suse.de>
Cc: Sahara Project <saharaproj@linsyssoft.com>, Christopher Harrer <charrer@alacritech.com>
Message-ID: <alpine.LFD.2.00.0903201736110.11454@localhost.localdomain>


Fix a softlock warning message thrown up by the rtnl_timer.
This was observed during the LTP tests (interface up/down test)

Signed-off-by: LinSysSoft Sahara Team <saharaproj@linsyssoft.com>
Signed-off-by: Mithlesh Thukral <mithlesh@linsyssoft.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/sxg/sxg.c |    2 --
 1 file changed, 2 deletions(-)

--- a/drivers/staging/sxg/sxg.c
+++ b/drivers/staging/sxg/sxg.c
@@ -2258,7 +2258,6 @@ int sxg_second_open(struct net_device * 
         /* Re-enable interrupts */
         SXG_ENABLE_ALL_INTERRUPTS(adapter);
 
-	netif_carrier_on(dev);
 	sxg_register_intr(adapter);
 	spin_unlock_irqrestore(&sxg_global.driver_lock, sxg_global.flags);
 	return (STATUS_SUCCESS);
@@ -2329,7 +2328,6 @@ static int sxg_entry_halt(struct net_dev
 	/* Disable interrupts */
 	SXG_DISABLE_ALL_INTERRUPTS(adapter);
 
-	netif_carrier_off(dev);
 	spin_unlock_irqrestore(&sxg_global.driver_lock, sxg_global.flags);
 
 	sxg_deregister_interrupt(adapter);
